app/src/middleware/authentication.js

```diff
@@ -116,7 +116,15 @@ const currentUser = async (req, res, next) => {
           throw new Error('OIDC environment variable KC_PUBLICKEY or keycloak.publicKey must be defined');
         }
 
-        if (isValid) {
+        // ensure token contains supported idp claim
+        const decodedToken = jwt.decode(bearerToken);
+        let idpSupported = true;
+        if (config.has('keycloak.identityKey')) {
+          const supportedIdentityKeys = config.get('keycloak.identityKey').split(',');
```

Comment:
IMO checking the `identity_provider` claim is a bit neater, but:
1. This would require a new environment variable (or that we re-purpose `keycloak.identityKey`).
2. This doesn't quite work for BCSC in the future because it doesn't even have an `identity_provider` claim.

I *might* be thinking a bit too far ahead with (2) though.